python/pyarrow/tests/test_dataset.py

```diff
@@ -549,7 +549,8 @@ def test_partitioning():
             pa.field('key', pa.float64())
         ])
     )
-    assert len(partitioning.dictionaries) == 0
+    assert len(partitioning.dictionaries) == 2
+    assert all(x is None for x in partitioning.dictionaries) is True
```

Comment:
```suggestion
    assert all(x is None for x in partitioning.dictionaries)
```

One minor comment: when doing an `assert all(...)`, the `is True` is not needed (`all` will return True or False, and `assert` will check that it is True)

Reply:
Made the change. Thanks for the suggestion!